### DISCUSSION TITLE
GLIBCXX_3.4.29 error

### DISCUSSION THREAD

**Discussion by jharamesh** [RosettaCommons/rosetta#234] [General]
Hi,
There has been some updates in our cluster and we started to see failure in our Rosetta runs with the following error.

'/opt/cray/pe/gcc-libs/libstdc++.so.6: version GLIBCXX_3.4.29' not found error

I performed static compilation of Rosetta, but the error remained.
Any suggestion, how this can be fixed.

**Comment by lyskov** [comment]
Something seems to be wrong, - if you indeed building static executable then it should not depend on shared libraries. I would recommend to double check most obvious things:
* check that you indeed building static executable, your build command line should be something like this: `python3 ./scons.py bin cxx=gcc extras=static mode=release -j64` (note the `extras=static`)
* double check that you using executables from the build that you did. Note that static build will be places somewhere in source/build dir with last part of the path being `static`. For instance on Ubuntu-18.04 with default compilers installed it will be something like `source/build/src/release/linux/6.11/64/x86/gcc/7/static`
* You can also run `ldd <rosetta-executable>` on your final executable, - if your build is indeed static then it should report something like `not a dynamic executable`

Hope this helps,
